tests/Unit/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Test_BjorhusImpl.cpp

```diff
@@ -0,0 +1,157 @@
+#include "Utilities/TMPL.hpp"
+
+namespace {
+using frame = Frame::Inertial;
+constexpr size_t VolumeDim = 3;
+
+// Test boundary conditions on dt<VSpacetimeMetric>
+void test_constraint_preserving_bjorhus_v_psi(
+    const size_t grid_size_each_dimension) noexcept {
+  // Setup grid
+  Mesh<VolumeDim> mesh{grid_size_each_dimension, Spectral::Basis::Legendre,
+                       Spectral::Quadrature::GaussLobatto};
+  const Direction<VolumeDim> direction(1, Side::Upper);  // +y direction
+  const size_t slice_grid_points =
+      mesh.extents().slice_away(direction.dimension()).product();
+
+  // Populate various tensors needed to compute BcDtVSpacetimeMetric
+  tnsr::iaa<DataVector, VolumeDim, frame> local_three_index_constraint(
+      slice_grid_points, std::numeric_limits<double>::signaling_NaN());
+  tnsr::I<DataVector, VolumeDim, frame> local_unit_interface_normal_vector(
+      slice_grid_points, std::numeric_limits<double>::signaling_NaN());
+  tnsr::aa<DataVector, VolumeDim, frame> local_char_projected_rhs_dt_v_psi(
+      slice_grid_points, std::numeric_limits<double>::signaling_NaN());
+  std::array<DataVector, 4> local_char_speeds{
+      DataVector(slice_grid_points,
+                 std::numeric_limits<double>::signaling_NaN()),
+      DataVector(slice_grid_points,
+                 std::numeric_limits<double>::signaling_NaN()),
+      DataVector(slice_grid_points,
+                 std::numeric_limits<double>::signaling_NaN()),
+      DataVector(slice_grid_points,
+                 std::numeric_limits<double>::signaling_NaN())};
+  {
+    // Setting 3idxConstraint
+    for (size_t i = 0; i < slice_grid_points; ++i) {
+      for (size_t a = 0; a <= VolumeDim; ++a) {
+        for (size_t b = 0; b <= VolumeDim; ++b) {
+          local_three_index_constraint.get(0, a, b)[i] = 11. - 3.;
+          local_three_index_constraint.get(1, a, b)[i] = 13. - 5.;
+          local_three_index_constraint.get(2, a, b)[i] = 17. - 7.;
+        }
+      }
+    }
+    // Setting unit_interface_normal_Vector
+    for (size_t i = 0; i < slice_grid_points; ++i) {
+      get<0>(local_unit_interface_normal_vector)[i] = -1.;
+      get<1>(local_unit_interface_normal_vector)[i] = 0.;
+      get<2>(local_unit_interface_normal_vector)[i] = 0.;
+    }
+    // Setting local_RhsVSpacetimeMetric
+    for (size_t i = 0; i < slice_grid_points; ++i) {
+      for (size_t a = 0; a <= VolumeDim; ++a) {
+        local_char_projected_rhs_dt_v_psi.get(0, a)[i] = 23.;
+        local_char_projected_rhs_dt_v_psi.get(1, a)[i] = 29.;
+        local_char_projected_rhs_dt_v_psi.get(2, a)[i] = 31.;
+        local_char_projected_rhs_dt_v_psi.get(3, a)[i] = 37.;
+      }
+    }
+    // Setting char speeds
+    for (size_t i = 0; i < slice_grid_points; ++i) {
+      local_char_speeds.at(0)[i] = -0.3;
+      local_char_speeds.at(1)[i] = -0.1;
+    }
+  }
+
+  // Allocate memory for output
+  tnsr::aa<DataVector, VolumeDim, frame> local_bc_dt_v_psi(
```

Comment:
Just so I understand, why is this not done with `make_with_value` here?

Reply:
No particular reason. We already know the number of points in the slice, so the first tensor has to be initialized directly with its constructor. All following initialization are done the same way just for consistency. I though we prefer `make_with_value` when we don't want to OR have the grid size explicitly.